### PR TITLE
Assignment1: Tyler + Austin

### PR DESCRIPTION
**Pull request by twoody0**
Partnered with @AustinH5544

### CODE REVIEW

**Review by BDullanty — COMMENTED** [review]
Looks good! Some ideas are in feedback

Pull Request title includes "Assignment 1" in the title ✔
Partner has also made a commit ✔
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔

Extra Credit
Come up with a new feature for the application:

Issue 3: Add a new feature request implemented ✔
Issue 3: Feature request unit tested ✔

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 1]
```diff
@@ -4,24 +4,49 @@ public class Program
...
+                return true;
+            }
+        }
+        return false;
     }
 
     public static string GetPercentCorrect(int numberCorrectAnswers, int numberOfQuestions)
     {
-        return (numberCorrectAnswers / numberOfQuestions * 100) + "%";
+        double percent = (numberCorrectAnswers / (double)numberOfQuestions * 100);
```
[the remark this answers is not in the document]
Comment (twoody0) [reply PrincessBrideTrivia/PrincessBrideTrivia/Program.cs]: Thanks for the suggestion, I added that to our program per your recommendation!

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 2]
```diff
@@ -87,4 +89,58 @@ private static void GenerateQuestionsFile(string filePath, int numberOfQuestions
...
+        //Act
+        Console.SetIn(userInput);
+        bool result = Program.ResetGame();
+
+        //Assert
+        Assert.IsFalse(result);
+    }
+
+    [TestMethod]
+    public void LoadQuestions_RandomizesQuestions()
```
[the remark this answers is not in the document]
Comment (twoody0) [reply PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs]: Thanks for the suggestion! When I added a recent push, this exact edge case did fail! But since changing from 10 questions to 100, this hasn't happened.

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 3]
```diff
@@ -87,4 +89,58 @@ private static void GenerateQuestionsFile(string filePath, int numberOfQuestions
...
+        //Act
+        Console.SetIn(userInput);
+        bool result = Program.ResetGame();
+
+        //Assert
+        Assert.IsTrue(result);
+    }
+
+    [TestMethod]
+    public void EndingTheGame_ReturnsFalseIfCorrect()
```
[the remark this answers is not in the document]
Comment (twoody0) [reply PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs]: I changed the name of the test method to your suggestion, and I also added a test method that uses Data Rows as well, thank you for the recommendation!

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 4]
```diff
@@ -87,4 +89,58 @@ private static void GenerateQuestionsFile(string filePath, int numberOfQuestions
             File.AppendAllLines(filePath, lines);
         }
     }
+
+    [TestMethod]
+    public void RessetingTheGame_ReturnsTrueIfCorrect()
```
[the remark this answers is not in the document]
Comment (twoody0) [reply PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs]: Thank you for the suggestion! I implemented the naming convention per your suggestion.

**Comment by ccornell88** [comment]
Good Work! Couldn't find much to change without getting picky.

Pull Request title includes "Assignment 1" in the title ✔
Partner has also made a commit ✔
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔

Extra Credit
Come up with a new feature for the application:

Issue 3: Add a new feature request implemented ✔
Issue 3: Feature request unit tested ✔

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Joshua-Lester3) [reply PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs]: @twoody0 I don't think it's good practice to have a test that could possibly fail, when nothing has changed - even when the chance is really small. My thought is passing in a seed into the Random constructor, perhaps through a parameter for the function. Also, here's [a good Stack Overflow post](https://stackoverflow.com/questions/3812154/unit-testing-a-method-with-random-behaviour) I found, though it may be a little ahead of what the class has gone over.

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 5]
```diff
@@ -87,4 +90,77 @@ private static void GenerateQuestionsFile(string filePath, int numberOfQuestions
...
+                secondLoad.Select(q => q.Text)
+            );
+
+            // Assert
+            Assert.IsTrue(areDifferent, "The order of questions should be different between loads.");
+        }
+        finally
+        {
+            File.Delete(filePath);
+        }
```
Comment (BenjaminMichaelis) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:164]: Nice job with using this to clean up your test files!

**Review by BenjaminMichaelis — COMMENTED** [review]
Nice!
